src/tools/plan-generator.ts

```diff
@@ -88,9 +88,12 @@ export class PlanGenerator {
     }
   }
 
-  /** Generates a Kotlin `StorageKey` from a StorageKey. */
-  createStorageKey(storageKey: StorageKey | undefined): string {
-    return `StorageKeyParser.parse("${(storageKey || '').toString()}")`;
+  /** Generates a Kotlin `StorageKey` from a recipe Handle. */
+  createStorageKey(handle: Handle): string {
+    switch (handle.fate) {
+      case 'create': return ktUtils.applyFun('CreateableStorageKey', [quote(handle.id)]);
```

Comment:
shouldn't you be checking for long running arc here? `create` handles of long running arcs should have keys assigned by the recipe2plan, not created by allocator.

Reply:
I'm pretty sure we throw an error earlier in the process -- before it reaches generation. 
I can add a `throw` here, or double check (i.e. add unit tests) to ensure that this is the case.

Reply:
Notes from meeting
-  check if the storage key is null, then we create the `CreateableStorageKey`. if it's not null, then we do the default.